scipy/stats/_continuous_distns.py

```diff
@@ -3639,8 +3697,319 @@ def alphanot1func(alpha, beta, TH, aTH, bTH, cosTH, tanTH, W):
+
+                with np.errstate(all="ignore"):
+                    intg_max = optimize.minimize_scalar(lambda theta: -f(theta), bounds=[-xi, np.pi/2])
+                    intg = integrate.quad(f, -xi, np.pi/2, points=[intg_max.x])[0]
+                    return alpha * intg / np.pi / np.abs(alpha-1) / (x0-zeta)
+            elif x0 == zeta:
+                return sc.gamma(1+1/alpha)*np.cos(xi)/np.pi/((1+zeta**2)**(1/alpha/2))
+            else:
+                return levy_stable_gen._pdf_single_value_zolotarev(-x, alpha, -beta)
+        else:
+            # since location zero, no need to reposition x for S_0 parameterization
+            xi = np.pi/2
+            if beta != 0:
+                warnings.warn('Density calculation unstable for alpha=1 and beta!=0. Use quadrature method instead.', RuntimeWarning)
+                                        
+                def V(theta):
+                    expr_1 = np.pi/2+beta*theta
+                    return 2. * expr_1 * np.exp(expr_1*np.tan(theta)/beta) / np.cos(theta) / np.pi 
+            
+                def g(theta):
+                    return np.exp(-np.pi * x / 2. / beta) * V(theta)
+                
+                def f(theta):
+                    return g(theta) * np.exp(-g(theta))
+            
+                with np.errstate(all="ignore"):
+                    intg_max = optimize.minimize_scalar(lambda theta: -f(theta), bounds=[-np.pi/2, np.pi/2])
+                    intg = integrate.fixed_quad(f, -np.pi/2, intg_max.x)[0] + integrate.fixed_quad(f, intg_max.x, np.pi/2)[0]
+                    return intg / np.abs(beta) / 2.                   
+            else:
+                return 1/(1+x**2)/np.pi
+
+    @staticmethod
+    def _cdf_single_value_zolotarev(x, alpha, beta):
+        """Calculate cdf using Zolotarev's methods as detailed in [BS].
+        """
+        zeta = -beta*np.tan(np.pi*alpha/2.)
+        if alpha != 1:
+            x0 = x + zeta  # convert to S_0 parameterization
+            xi = np.arctan(-zeta)/alpha
+            
+            def V(theta):
+                return np.cos(alpha*xi)**(1/(alpha-1)) * \
+                                (np.cos(theta)/np.sin(alpha*(xi+theta)))**(alpha/(alpha-1)) * \
+                                (np.cos(alpha*xi+(alpha-1)*theta)/np.cos(theta))
+            if x0 > zeta:
+                c_1 = 1 if alpha > 1 else .5 - xi/np.pi
+                
+                def f(theta):
+                    return np.exp(-V(theta)*(x0-zeta)**(alpha/(alpha-1)))
+
+                with np.errstate(all="ignore"):
+                    intg = integrate.quad(f, -xi, np.pi/2)[0]
+                    return c_1 + np.sign(1-alpha) * intg / np.pi
+            elif x0 == zeta:
+                return .5 - xi/np.pi
+            else:
+                return 1 - levy_stable_gen._cdf_single_value_zolotarev(-x, alpha, -beta)
+                
+        else:
```

Comment:
This whole branch doesn't have test coverage (see [coverage report](https://codecov.io/gh/scipy/scipy/compare/b582245fe89dd7363f650b3da8124929b39ce13a...910a63f0de8704f1d95c0ef565823e2c21b4c737/src/scipy/stats/_continuous_distns.py?before=scipy/stats/_continuous_distns.py)), could you try to construct a test that exercises this code?

Reply:
covered now